ksp_plugin_adapter/time_span.cs

```diff
@@ -90,30 +90,28 @@ class PrincipiaTimeSpan {
   public double total_seconds => seconds_;
 
   public static bool TryParse(string text,
-                              bool with_seconds,
                               out PrincipiaTimeSpan time_span) {
     time_span = new PrincipiaTimeSpan(double.NaN);
     // Using a technology that is customarily used to parse HTML.
-    string pattern = @"^[+]?\s*(\d+)\s*" +
+    string pattern = @"^[+]?\s*(?<days>\d+)\s*" +
                      day_symbol +
-                     @"\s*(\d+)\s*h\s*(\d+)\s*min";
-    if (with_seconds) {
-      pattern += @"\s*([0-9.,']+)\s*s$";
-    } else {
-      pattern += @"$";
-    }
+                     @"\s*" +
+                     @"(?:(?<hours>\d+)\s*h\s*" +
+                     @"(?:(?<minutes>\d+)\s*min\s*"+
+                     @"(?:(?<seconds>[0-9.,']+)\s*s)?)?)?$";
     var regex = new Regex(pattern);
     var match = regex.Match(text);
     if (!match.Success) {
       return false;
     }
-    string days = match.Groups[1].Value;
-    string hours = match.Groups[2].Value;
-    string minutes = match.Groups[3].Value;
-    string seconds = "0";
-    if (with_seconds) {
-      seconds = match.Groups[4].Value;
-    }
+
+    string days = match.Groups["days"].Value;
```

Comment:
Shouldn’t the days be optional as well ? Having to write "5 h" as "0 d6 5 h" seems annoying.

Reply:
Updated the regexp so that all components are optional, example [here](http://regexstorm.net/tester?p=%5e%5b%2b%5d%3f%5cs*%28%3f%3a%28%3f%3cdays%3e%5cd%2b%29%5cs*d6%5cs*%29%3f%28%3f%3a%28%3f%3chours%3e%5cd%2b%29%5cs*h%5cs*%29%3f%28%3f%3a%28%3f%3cminutes%3e%5cd%2b%29%5cs*min%5cs*%29%3f%28%3f%3a%28%3f%3cseconds%3e%5b0-9.%2c%27%5d%2b%29%5cs*s%5cs*%29%3f%24&i=1+d6+3h+17+s++).  Also fix a glitch where the data was not reformatted if it was numerically identical.